tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions.dart

```diff
@@ -324,6 +324,19 @@ class Restrictions {
           )
         ];
       }
+
+      // This is temporary until [PartAllocator] and [PartOfAllocator] can automatically allocate the imports from the sub-classes relative to the parent imports.
+      if (currentModel.sealedTopNode != null) {
+        if (currentModel.subDirParts.join() !=
+            currentModel.sealedTopNode!.subDirParts.join()) {
+          return [
+            SourceSpanSeverityException(
+              'All models in a sealed library must be in the same subdirectory. The class "${currentModel.className}" needs to be located in the same subdirectory as "${currentModel.sealedTopNode!.className}".',
```

Comment:
Suggestion: Please remove the dangerous deref `!`.

Reply:
Suggestion: Please validate the value exists before using it.

Assign `currentModel.sealedTopNode` to a variable outside and nullcheck that variable instead of fetching it with the getter multiple times.